### PR TITLE
jwt authorizaion

### PR DESCRIPTION
**Pull request by fredie7**
### Summary
This PR adds a JWT verification middleware. The middleware is used for protecting or private routes by authorising requests using the token provided in authorisation header.

### CODE REVIEW

File: controllers/auth.js [hunk 1]
```diff
@@ -37,11 +37,31 @@ const authController = {
...
+    //   expiresIn: jwtExpiryTime,
+    // });
+    if (!token) {
+      return res.status(403).json({ error: 'user unauthorized' });
+    }
+    jwt.verify(token, jwtSecretKey, (err, authData) => {
+      if (err) {
+        res.status(403).json({ error: 'invalid token' });
+      } else {
+        res.json({ authData });
```
Comment (jsamchineme) [inline controllers/auth.js]: The authData is not to be returned in a response. It is passed on to the next middleware, attached to the request object. 
Just before your `next()` statement, do `req.decoded = authData`

File: routers/auth.js [hunk 2]
```diff
@@ -5,9 +5,9 @@ import validator from '../validator';
 const router = express.Router();
 
 const { signinValidator, signupValidator } = validator;
-const { signup, signin } = authController;
+const { signup, signin, verifyUser } = authController;
 
 router.post('/signup', signupValidator, signup);
-router.post('/signin', signinValidator, signin);
+router.post('/signin', signinValidator, verifyUser, signin);
```
Comment (jsamchineme) [inline routers/auth.js]: You can't have been hr verifyUser middleware before your signing. You use if for protected routes eg. CreateParcel that requires a valid user's token.

**Review by jsamchineme — CHANGES_REQUESTED** [review]

File: routers/parcel.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import authController from '../controllers/auth';
+
+const router = express.Router();
+
+router.post('/protected', authController.verifyToken, (req, res) => {
```
Comment (jsamchineme) [inline routers/parcel.js]: With what you have here, you should be able to test on Postman that the middleware works.
Make a POST request to `api/v1/parcels/protected` providing the token in the authorization header, and see what you get returned. Use both a valid token and an invalid token. A valid token is the one you get after sign-in, untampered. An invalid would be any other bunch characters, or a token that has expired.

File: app.js [hunk 4]
```diff
@@ -14,8 +14,11 @@ app.use(morgan('dev'));
 app.use(expressValidator());
 
 const authRoute = require('./routers/auth');
```
Comment (jsamchineme) [inline app.js]: Remove all `require()` statements. replace them with the `import` format.

File: routers/parcel.js (hunk 3 above)
Comment (fredie7) [reply routers/parcel.js]: it says unauthhorized

File: app.js (hunk 4 above)
Comment (fredie7) [reply app.js]: I've fixed those

Comment (jsamchineme) [inline app.js]: fix this too.

File: app.js [hunk 5]
```diff
@@ -14,8 +14,11 @@ app.use(morgan('dev'));
...
+const parcelRoutes = require('./routers/parcel');
```
Comment (jsamchineme) [inline app.js]: fix this too

**Review by jsamchineme — COMMENTED** [review]
Fix the rest of the require statements and we should be good.

**Pull request merged**